### ISSUE TITLE
feat: Add colour to overscrolling bounce

### ISSUE THREAD

**Issue by devleaf-matt** [ionic-team/ionic#19940]
# Feature Request

**Ionic version:**

[x] **4.x**

**Describe the Feature Request**

To be able to change a directional overscroll colour. 

**Describe Preferred Solution**

On iOS or any platform where overscroll bounce is enabled. You should be able to change the overscroll colour such that it allows you to match anything within the immediate vicinity. 

**Describe Alternatives**

None that really work. I've tried setting the ion-content colour to being different from an element inside but due to the nature of this, when overscrolling downwards the page goes the overscroll colour the same as the top. I've also tried adding a fixed element which sits above ion-content to overscroll to, however, it seems like overscrolling is part of the ion-content.

**Related Code**

At the moment I am trying to use the iOS 11 style headers and am encountering a problem by having my header colour different to the `<ion-content>`. Here is the current code: 

```
<ion-header>
  <ion-toolbar color="primary"></ion-toolbar>

  <!-- I have code here which is what I used to use to simulate an iOS 11 header, but won't paste in this example -->
</ion-header>

<ion-content>
  <ion-header collapse="content">
    <ion-toolbar color="primary">
      <ion-searchbar></ion-searchbar>
    </ion-toolbar>
  </ion-header>
</ion-content>
```

**Additional Context**

![RPReplay_Final1574115578](https://user-images.githubusercontent.com/49694881/69099490-c233eb80-0a52-11ea-9852-59474a942f42.gif)

Theoretically what would be nice is to define an attribute like `overscroll-top="primary"` on the `<ion-content>` or something similar if theoretically possible. As can be seen by the video above. When scrolling up it would be nice to match the orange colour when overscrolling upwards but maintain the white background when overscrolling down. 

![RPReplay_Final1574118446](https://user-images.githubusercontent.com/49694881/69101788-71bf8c80-0a58-11ea-969b-61d7680df9f7.gif)

Here is Hole19 which is the only app I have on my phone showing exactly what i'd like to be able to do. At the top, the blue overscroll segment continues but at the bottom the white color overscrolls.

**Comment by liamdebeasi** [comment]
Thanks for the issue. I am going to close this as this is already possible in Ionic Framework. I created a CodePen here: https://codepen.io/liamdebeasi/pen/dyyazbe?ionic:mode=ios

You might need to tweak the colors/properties a bit to match your use case.

**Comment by devleaf-matt** [comment]
Hi Liam,

Sorry to re-open, but is there any way to do this without the `<ion-refresher>`. For example if I wanted the ability to get the colour on the `<ion-title size="large">` as updated in a recent version of ionic, as at the moment it causes the issue as seen in the screenshot of my app. The hole 19 screenshot was just to demonstrate how the overscroll at the top was a different colour to the overscroll at the bottom.

**Comment by liamdebeasi** [comment]
You could likely just put a div with the color you wanted in the `fixed` slot. (That's all ion-refresher is doing in terms of layout/placement).

So something like this:

```
<div class="custom-color" slot="fixed"></div>
```

**Comment by devleaf-matt** [comment]
Hmm. Not having any luck with the `slot="fixed"` div. Still having the same issue as before. The closest I've gotten so far is to have a refresher (as your codepen demonstrates) with a huge `pull-min` and large height, as well as masking the ion-icons. However, the overflow still makes its way through looking like it's glitchy. 

Code: 

```
<ion-header>
  <ion-toolbar color="primary">
    <ion-title>Lists</ion-title>
  </ion-toolbar>
</ion-header>

<ion-content>
  <!-- <ion-refresher class="overscroller" slot="fixed" pull-min="99999999">
    <ion-refresher-content></ion-refresher-content>
  </ion-refresher> -->

  <ion-header>
    <ion-toolbar color="primary">
        <ion-searchbar></ion-searchbar>
    </ion-toolbar>
  </ion-header>
</ion-content>
```

CSS: 

```
ion-refresher {
  background: var(--ion-color-primary);
  height: 200px;
}

ion-refresher-content {
  height: 60px;
  ion-icon {
    color: var(--ion-color-primary);
  }
}

ion-content {
  --background: var(--ion-color-primary);
}

ion-toolbar {
  --border-width: 0px !important;
}
```

Does this on my live device (iOS): 

![RPReplay_Final1574182412](https://user-images.githubusercontent.com/49694881/69170439-fc07ff00-0af1-11ea-98f5-fb1aa8969cf3.gif)
